src/reasoner/soma_dfl/soma_dfl.py

```diff
@@ -0,0 +1,144 @@
+        self.fnMap = {self.hasDisposition: lambda p, s, o, bounding: self._evaluateSimpleGoal(p, s, o, bounding, sr.dl.whatDispositionsDoesObjectHave, sr.dl.whatHasDisposition),
+                      self.hasPart: lambda p, s, o, bounding: self._evaluateSimpleGoal(p, s, o, bounding, sr.dl.whatPartTypesDoesObjectHave, sr.dl.whatHasPartType),
+                      self.hasConstituent: lambda p, s, o, bounding: self._evaluateSimpleGoal(p, s, o, bounding, sr.dl.whatConstituentsDoesObjectHave, sr.dl.whatHasConstituent)}
+        self.classes = set()
+        # TODO: UseMatch: task, instrument, object acted on
+        
+    def _evaluateSimpleGoal(self, p, s, o, bounding, fnOVar, fnSVar):
+        if ((False, True) == bounding):
+            todo = self._ensureIndividual2Classes(s)
+            dispositions = set().union(*[fnOVar(str(c)) for c in todo])
+            _ = [goal.push(Bindings({o: IRIAtom(sr.dl.expandName(d))})) for d in dispositions]
+        else:
+            concepts = [sr.dl.expandName(c) for c in fnSVar(str(o))]
+            instances = set.union(*[self._ensureClass2Individuals(c) for c in concepts])
+            if ((True, False) == bounding):
+                _ = [goal.push(Bindings({s: IRIAtom(sr.dl.expandName(i))})) for i in instances]
+            elif ((False, False) == bounding) and str(s) in instances:
+                goal.push(Bindings({}))
+        return True
+
+    def _ensureIndividual2Classes(self, entity):
+        entity = str(entity)
+        entity = sr.dl.expandName(entity)
+        retq = [entity]
+        if entity not in self.classes:
+            query_term = GraphSequence([GraphPattern(IRIAtom(entity), self.rdfType, Variable("z"))])
+            retq = []
+            self.storage().query(GraphQuery(query_term), lambda bindings : retq.append(bindings.get(Variable("z"))))
+            retq = list(set().union(*[sr.dl.whatSuperclasses(str(c)) for c in retq]))
+        return retq
+        
+    def _ensureClass2Individuals(self, entity):
+        entity = str(entity)
+        entity = sr.dl.expandName(entity)
+        retq = [entity]
+        if entity in self.classes:
+            retq = set()
+            for sc in sr.dl.whatSubclasses(str(entity)):
+                query_term = GraphSequence([GraphPattern(Variable("z"), self.rdfType, IRIAtom(sc))])
+                self.storage().query(GraphQuery(query_term), lambda bindings : retq.add(str(bindings.get(Variable("z")))))
+            retq = list(retq)
+        return retq
+                
+    def initializeReasoner(self, config: PropertyTree) -> bool:
+        # TODO: MAJOR: separate cache, namespace etc. for each instance of this class, right now it is only allowed to be a singleton
+        # TODO: parse and use config.
+        # Flags that would make sense:
+        #     ontology and usematch files to use
```

Comment:
there is a mechanism dedicated for loading input files into a reasoner.

In settings files, you can have an import section like here:
https://github.com/knowrob/knowrob/blob/dev/tests/settings/lpn.json#L28

**"format": "prolog"** is important. Now in the reasoner, you can tell KnowRob about that your reasoner can handle specific formats like the Prolog reasoner does:
```
	addDataHandler("prolog", [this]
			(const DataSourcePtr &dataFile) { return consult(dataFile->uri()); });
```

Reply:
Might not be covered in unit tests in the pre-release, so could be that the GIL crashes when you try from Python.